2022-09-07 03:11:05 info: ---------------------------------------
2022-09-07 03:11:05 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-09-07 03:11:05 info: ---------------------------------------
2022-09-07 03:11:09 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (3 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls
    ✓ should call `new Promise()`
    ✓ should take a single argument (1 ms)
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument
    ✕ should resolve when called with a string argument (7 ms)
    ✕ should reject with an Error object when called without an argument

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

  ● getAnonName › should resolve when called with a string argument

    TypeError: getAnonName is not a function

      75 |       .spyOn(global, 'setTimeout')
      76 |       .mockImplementation((cb) => cb());
    > 77 |     const promise = getAnonName('John');
         |                     ^
      78 |     timeoutSpy.mockRestore();
      79 |     expect(promise).toBeInstanceOf(Promise);
      80 |     return expect(promise).resolves.toEqual('John Doe');

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:77:21)

  ● getAnonName › should resolve when called with a string argument

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      70 |
      71 |   test('should resolve when called with a string argument', async () => {
    > 72 |     expect.assertions(3);
         |            ^
      73 |     expect(exported).toBeDefined();
      74 |     const timeoutSpy = jest
      75 |       .spyOn(global, 'setTimeout')

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:72:12)

  ● getAnonName › should reject with an Error object when called without an argument

    TypeError: getAnonName is not a function

      87 |       .spyOn(global, 'setTimeout')
      88 |       .mockImplementation((cb) => cb());
    > 89 |     const promise = getAnonName();
         |                     ^
      90 |     timeoutSpy.mockRestore();
      91 |     expect(promise).toBeInstanceOf(Promise);
      92 |     return expect(promise).rejects.toBeInstanceOf(Error);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:89:21)

  ● getAnonName › should reject with an Error object when called without an argument

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      82 |
      83 |   test('should reject with an Error object when called without an argument', async () => {
    > 84 |     expect.assertions(3);
         |            ^
      85 |     expect(exported).toBeDefined();
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:84:12)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 6 passed, 9 total
Snapshots:   0 total
Time:        1.02 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-09-07 03:11:11 error: *** ESLint Report ***

ex1-johnWho.js
  12:7  error    'getAnonName' is assigned a value but never used  no-unused-vars
  24:1  warning  Remove commented-out code                         hyf/no-commented-out-code
  29:1  warning  Remove commented-out code                         hyf/no-commented-out-code

✖ 3 problems (1 error, 2 warnings)


2022-09-07 03:11:55 info: ---------------------------------------
2022-09-07 03:11:55 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-09-07 03:11:55 info: ---------------------------------------
2022-09-07 03:11:58 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls (1 ms)
    ✓ should call `new Promise()` (1 ms)
    ✓ should take a single argument
    ✓ `resolve()` should be called with a one argument (1 ms)
    ✓ `reject()` should be called with a one argument
    ✓ should resolve when called with a string argument (3 ms)
    ✓ should reject with an Error object when called without an argument (11 ms)

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.802 s, estimated 1 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-09-07 03:12:44 info: ---------------------------------------
2022-09-07 03:12:44 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-09-07 03:12:44 info: ---------------------------------------
2022-09-07 03:12:48 info: All unit tests passed.
2022-09-07 03:12:52 info: All steps were completed successfully
2022-09-07 03:14:27 info: -------------------------------------------------
2022-09-07 03:14:27 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-09-07 03:14:27 info: -------------------------------------------------
2022-09-07 03:14:27 info: Exercise has not yet been modified
2022-09-07 03:14:30 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `checkDoubleDigits` should not contain unneeded console.log calls (1 ms)
    ✕ should call new Promise() (1 ms)
    ✕ `resolve()` should be called with a one argument (2 ms)
    ✕ `reject()` should be called with a one argument
    ✕ should be a function that takes a single argument (1 ms)
    ✕ (9) should return a rejected promise with an Error object (1 ms)
    ✕ (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (99) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (100) should return a rejected promise with an Error object

  ● checkDoubleDigits › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

  ● checkDoubleDigits › should call new Promise()

    expect(received).toBeDefined()

    Received: undefined

      43 |
      44 |   test('should call new Promise()', () => {
    > 45 |     expect(state.newPromise).toBeDefined();
         |                              ^
      46 |   });
      47 |
      48 |   test('`resolve()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:45:30)

  ● checkDoubleDigits › `resolve()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      47 |
      48 |   test('`resolve()` should be called with a one argument', () => {
    > 49 |     expect(state.resolve).toBe(1);
         |                           ^
      50 |   });
      51 |
      52 |   test('`reject()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:49:27)

  ● checkDoubleDigits › `reject()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      51 |
      52 |   test('`reject()` should be called with a one argument', () => {
    > 53 |     expect(state.reject).toBe(1);
         |                          ^
      54 |   });
      55 |
      56 |   test('should be a function that takes a single argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:53:26)

  ● checkDoubleDigits › should be a function that takes a single argument

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      58 |     expect(
      59 |       typeof checkDoubleDigits === 'function' && checkDoubleDigits.length === 1
    > 60 |     ).toBe(true);
         |       ^
      61 |   });
      62 |
      63 |   test('(9) should return a rejected promise with an Error object', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:60:7)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Promise

    Received value has no prototype
    Received value: undefined

      64 |     expect.assertions(2);
      65 |     const promise = checkDoubleDigits(9);
    > 66 |     expect(promise).toBeInstanceOf(Promise);
         |                     ^
      67 |     return expect(promise).rejects.toBeInstanceOf(Error);
      68 |   });
      69 |

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:66:21)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      62 |
      63 |   test('(9) should return a rejected promise with an Error object', () => {
    > 64 |     expect.assertions(2);
         |            ^
      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);
      67 |     return expect(promise).rejects.toBeInstanceOf(Error);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:64:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Promise

    Received value has no prototype
    Received value: undefined

      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
    > 74 |     expect(promise).toBeInstanceOf(Promise);
         |                     ^
      75 |     return expect(promise).resolves.toEqual(
      76 |       expect.stringContaining('This is a double digit number!')
      77 |     );

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:74:21)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Promise

    Received value has no prototype
    Received value: undefined

      81 |     expect.assertions(2);
      82 |     const promise = checkDoubleDigits(99);
    > 83 |     expect(promise).toBeInstanceOf(Promise);
         |                     ^
      84 |     return expect(promise).resolves.toEqual(
      85 |       expect.stringContaining('This is a double digit number!')
      86 |     );

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:83:21)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Promise

    Received value has no prototype
    Received value: undefined

      90 |     expect.assertions(2);
      91 |     const promise = checkDoubleDigits(100);
    > 92 |     expect(promise).toBeInstanceOf(Promise);
         |                     ^
      93 |     return expect(promise).rejects.toBeInstanceOf(Error);
      94 |   });
      95 | });

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:92:21)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      88 |
      89 |   test('(100) should return a rejected promise with an Error object', () => {
    > 90 |     expect.assertions(2);
         |            ^
      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);
      93 |     return expect(promise).rejects.toBeInstanceOf(Error);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:90:12)

Test Suites: 1 failed, 1 total
Tests:       9 failed, 2 passed, 11 total
Snapshots:   0 total
Time:        0.819 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2022-09-07 03:14:39 info: Disclaimer turned off
2022-09-09 03:49:02 info: -------------------------------------------------
2022-09-09 03:49:02 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-09-09 03:49:02 info: -------------------------------------------------
2022-09-09 03:49:06 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ `checkDoubleDigits` should not contain unneeded console.log calls (2 ms)
    ✓ should call new Promise()
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument
    ✓ should be a function that takes a single argument
    ✕ (9) should return a rejected promise with an Error object (4 ms)
    ✓ (10) should return a promise that resolves to "This is a double digit number!" (2 ms)
    ✓ (99) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (100) should return a rejected promise with an Error object (1 ms)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    expect(received).rejects.toBeInstanceOf(expected)

    Expected constructor: Error

    Received value has no prototype
    Received value: "Expected a double digit number but got 9"

      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);
    > 67 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |                                    ^
      68 |   });
      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {

      at Object.toBeInstanceOf (node_modules/expect/build/index.js:285:22)
      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:67:36)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    expect(received).rejects.toBeInstanceOf(expected)

    Expected constructor: Error

    Received value has no prototype
    Received value: "Expected a double digit number but got 100"

      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);
    > 93 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |                                    ^
      94 |   });
      95 | });
      96 |

      at Object.toBeInstanceOf (node_modules/expect/build/index.js:285:22)
      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:93:36)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 9 passed, 11 total
Snapshots:   0 total
Time:        0.807 s, estimated 1 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2022-09-09 03:51:44 info: -------------------------------------------------
2022-09-09 03:51:44 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-09-09 03:51:44 info: -------------------------------------------------
2022-09-09 03:51:57 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js (10.586 s)
  checkDoubleDigits
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed
    ✓ `checkDoubleDigits` should not contain unneeded console.log calls (1 ms)
    ✓ should call new Promise()
    ✓ `resolve()` should be called with a one argument
    ✕ `reject()` should be called with a one argument (2 ms)
    ✓ should be a function that takes a single argument
    ✕ (9) should return a rejected promise with an Error object (5006 ms)
    ✓ (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✓ (99) should return a promise that resolves to "This is a double digit number!"
    ✕ (100) should return a rejected promise with an Error object (5003 ms)

  ● checkDoubleDigits › `reject()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      51 |
      52 |   test('`reject()` should be called with a one argument', () => {
    > 53 |     expect(state.reject).toBe(1);
         |                          ^
      54 |   });
      55 |
      56 |   test('should be a function that takes a single argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:53:26)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    Expected a double digit number but got 9

      19 |     } else {
      20 |       Promise.reject(
    > 21 |         new Error(`Expected a double digit number but got ${num}`)
         |         ^
      22 |       );
      23 |     }
      24 |   });

      at 3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js:21:9
      at checkDoubleDigits (3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js:16:10)
      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:65:21)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      61 |   });
      62 |
    > 63 |   test('(9) should return a rejected promise with an Error object', () => {
         |   ^
      64 |     expect.assertions(2);
      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);

      at 3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:63:3
      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:10:1)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      62 |
      63 |   test('(9) should return a rejected promise with an Error object', () => {
    > 64 |     expect.assertions(2);
         |            ^
      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);
      67 |     return expect(promise).rejects.toBeInstanceOf(Error);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:64:12)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    Expected a double digit number but got 100

      19 |     } else {
      20 |       Promise.reject(
    > 21 |         new Error(`Expected a double digit number but got ${num}`)
         |         ^
      22 |       );
      23 |     }
      24 |   });

      at 3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js:21:9
      at checkDoubleDigits (3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js:16:10)
      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:91:21)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      87 |   });
      88 |
    > 89 |   test('(100) should return a rejected promise with an Error object', () => {
         |   ^
      90 |     expect.assertions(2);
      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);

      at 3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:89:3
      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:10:1)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      88 |
      89 |   test('(100) should return a rejected promise with an Error object', () => {
    > 90 |     expect.assertions(2);
         |            ^
      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);
      93 |     return expect(promise).rejects.toBeInstanceOf(Error);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:90:12)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 8 passed, 11 total
Snapshots:   0 total
Time:        10.648 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2022-09-09 03:51:58 error: *** ESLint Report ***

ex2-checkDoubleDigits.js
  16:32  error  'reject' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)


2022-09-09 03:52:37 info: -------------------------------------------------
2022-09-09 03:52:37 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-09-09 03:52:37 info: -------------------------------------------------
2022-09-09 03:52:40 info: All unit tests passed.
2022-09-09 03:52:43 info: All steps were completed successfully
2022-09-09 05:08:58 info: ---------------------------------------
2022-09-09 05:08:58 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:08:58 info: ---------------------------------------
2022-09-09 05:09:01 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✕ should call `new Promise()`
    ✕ `resolve()` should be called with a one argument (2 ms)
    ✕ `reject()` should be called with a one argument
    ✕ should resolve when the die settles successfully (1 ms)
    ✕ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

  ● rollDie › should call `new Promise()`

    expect(received).toBeDefined()

    Received: undefined

      38 |
      39 |   test('should call `new Promise()`', () => {
    > 40 |     expect(state.newPromise).toBeDefined();
         |                              ^
      41 |   });
      42 |
      43 |   test('`resolve()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:40:30)

  ● rollDie › `resolve()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      42 |
      43 |   test('`resolve()` should be called with a one argument', () => {
    > 44 |     expect(state.resolve).toBe(1);
         |                           ^
      45 |   });
      46 |
      47 |   test('`reject()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:44:27)

  ● rollDie › `reject()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      46 |
      47 |   test('`reject()` should be called with a one argument', () => {
    > 48 |     expect(state.reject).toBe(1);
         |                          ^
      49 |   });
      50 |
      51 |   test('should resolve when the die settles successfully', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:48:26)

  ● rollDie › should resolve when the die settles successfully

    TypeError: rollDie is not a function

      59 |       .mockImplementation((cb) => cb());
      60 |
    > 61 |     const promise = rollDie();
         |                     ^
      62 |     expect(promise).toBeInstanceOf(Promise);
      63 |     const assertionPromise = expect(promise).resolves.toBeDefined();
      64 |

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:61:21)

  ● rollDie › should resolve when the die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      50 |
      51 |   test('should resolve when the die settles successfully', () => {
    > 52 |     expect.assertions(3);
         |            ^
      53 |     expect(exported).toBeDefined();
      54 |
      55 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:52:12)

  ● rollDie › should reject with an Error when the die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      73 |
      74 |   test('should reject with an Error when the die rolls off the table', async () => {
    > 75 |     expect.assertions(3);
         |            ^
      76 |     expect(exported).toBeDefined();
      77 |
      78 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:75:12)

Test Suites: 1 failed, 1 total
Tests:       6 failed, 1 passed, 7 total
Snapshots:   0 total
Time:        0.857 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-09-09 05:09:03 error: *** ESLint Report ***

ex3-rollDie.js
  47:10  error    'main' is defined but never used  no-unused-vars
  59:1   warning  Remove commented-out code         hyf/no-commented-out-code

✖ 2 problems (1 error, 1 warning)


2022-09-09 05:15:57 info: ---------------------------------------
2022-09-09 05:15:57 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:15:57 info: ---------------------------------------
2022-09-09 05:16:01 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (3 ms)
    ✕ should call `new Promise()` (1 ms)
    ✕ `resolve()` should be called with a one argument (2 ms)
    ✕ `reject()` should be called with a one argument
    ✕ should resolve when the die settles successfully
    ✕ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

  ● rollDie › should call `new Promise()`

    expect(received).toBeDefined()

    Received: undefined

      38 |
      39 |   test('should call `new Promise()`', () => {
    > 40 |     expect(state.newPromise).toBeDefined();
         |                              ^
      41 |   });
      42 |
      43 |   test('`resolve()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:40:30)

  ● rollDie › `resolve()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      42 |
      43 |   test('`resolve()` should be called with a one argument', () => {
    > 44 |     expect(state.resolve).toBe(1);
         |                           ^
      45 |   });
      46 |
      47 |   test('`reject()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:44:27)

  ● rollDie › `reject()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      46 |
      47 |   test('`reject()` should be called with a one argument', () => {
    > 48 |     expect(state.reject).toBe(1);
         |                          ^
      49 |   });
      50 |
      51 |   test('should resolve when the die settles successfully', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:48:26)

  ● rollDie › should resolve when the die settles successfully

    TypeError: rollDie is not a function

      59 |       .mockImplementation((cb) => cb());
      60 |
    > 61 |     const promise = rollDie();
         |                     ^
      62 |     expect(promise).toBeInstanceOf(Promise);
      63 |     const assertionPromise = expect(promise).resolves.toBeDefined();
      64 |

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:61:21)

  ● rollDie › should resolve when the die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      50 |
      51 |   test('should resolve when the die settles successfully', () => {
    > 52 |     expect.assertions(3);
         |            ^
      53 |     expect(exported).toBeDefined();
      54 |
      55 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:52:12)

  ● rollDie › should reject with an Error when the die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      73 |
      74 |   test('should reject with an Error when the die rolls off the table', async () => {
    > 75 |     expect.assertions(3);
         |            ^
      76 |     expect(exported).toBeDefined();
      77 |
      78 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:75:12)

Test Suites: 1 failed, 1 total
Tests:       6 failed, 1 passed, 7 total
Snapshots:   0 total
Time:        0.806 s, estimated 1 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-09-09 05:16:02 error: *** ESLint Report ***

ex3-rollDie.js
  47:10  error    'main' is defined but never used  no-unused-vars
  58:1   warning  Remove commented-out code         hyf/no-commented-out-code

✖ 2 problems (1 error, 1 warning)


2022-09-09 05:23:46 info: ---------------------------------------
2022-09-09 05:23:46 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:23:46 info: ---------------------------------------
2022-09-09 05:23:50 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ should call `new Promise()`
    ✕ `resolve()` should be called with a one argument (1 ms)
    ✓ `reject()` should be called with a one argument
    ✕ should resolve when the die settles successfully (1 ms)
    ✕ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

  ● rollDie › `resolve()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      42 |
      43 |   test('`resolve()` should be called with a one argument', () => {
    > 44 |     expect(state.resolve).toBe(1);
         |                           ^
      45 |   });
      46 |
      47 |   test('`reject()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:44:27)

  ● rollDie › should resolve when the die settles successfully

    TypeError: rollDie is not a function

      59 |       .mockImplementation((cb) => cb());
      60 |
    > 61 |     const promise = rollDie();
         |                     ^
      62 |     expect(promise).toBeInstanceOf(Promise);
      63 |     const assertionPromise = expect(promise).resolves.toBeDefined();
      64 |

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:61:21)

  ● rollDie › should resolve when the die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      50 |
      51 |   test('should resolve when the die settles successfully', () => {
    > 52 |     expect.assertions(3);
         |            ^
      53 |     expect(exported).toBeDefined();
      54 |
      55 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:52:12)

  ● rollDie › should reject with an Error when the die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      73 |
      74 |   test('should reject with an Error when the die rolls off the table', async () => {
    > 75 |     expect.assertions(3);
         |            ^
      76 |     expect(exported).toBeDefined();
      77 |
      78 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:75:12)

Test Suites: 1 failed, 1 total
Tests:       4 failed, 3 passed, 7 total
Snapshots:   0 total
Time:        0.624 s, estimated 1 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-09-09 05:23:51 error: *** ESLint Report ***

ex3-rollDie.js
  18:11  error    'rollOnce' is assigned a value but never used  no-unused-vars
  30:9   error    'callback' is not defined                      no-undef
  44:10  error    'main' is defined but never used               no-unused-vars
  55:1   warning  Remove commented-out code                      hyf/no-commented-out-code

✖ 4 problems (3 errors, 1 warning)


2022-09-09 05:24:48 info: ---------------------------------------
2022-09-09 05:24:48 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:24:48 info: ---------------------------------------
2022-09-09 05:24:50 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ should call `new Promise()`
    ✕ `resolve()` should be called with a one argument (2 ms)
    ✓ `reject()` should be called with a one argument (1 ms)
    ✕ should resolve when the die settles successfully
    ✕ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › `resolve()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      42 |
      43 |   test('`resolve()` should be called with a one argument', () => {
    > 44 |     expect(state.resolve).toBe(1);
         |                           ^
      45 |   });
      46 |
      47 |   test('`reject()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:44:27)

  ● rollDie › should resolve when the die settles successfully

    TypeError: rollDie is not a function

      59 |       .mockImplementation((cb) => cb());
      60 |
    > 61 |     const promise = rollDie();
         |                     ^
      62 |     expect(promise).toBeInstanceOf(Promise);
      63 |     const assertionPromise = expect(promise).resolves.toBeDefined();
      64 |

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:61:21)

  ● rollDie › should resolve when the die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      50 |
      51 |   test('should resolve when the die settles successfully', () => {
    > 52 |     expect.assertions(3);
         |            ^
      53 |     expect(exported).toBeDefined();
      54 |
      55 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:52:12)

  ● rollDie › should reject with an Error when the die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      73 |
      74 |   test('should reject with an Error when the die rolls off the table', async () => {
    > 75 |     expect.assertions(3);
         |            ^
      76 |     expect(exported).toBeDefined();
      77 |
      78 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:75:12)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 4 passed, 7 total
Snapshots:   0 total
Time:        0.606 s, estimated 1 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-09-09 05:24:51 error: *** ESLint Report ***

ex3-rollDie.js
  18:11  error    'rollOnce' is assigned a value but never used  no-unused-vars
  43:10  error    'main' is defined but never used               no-unused-vars
  54:1   warning  Remove commented-out code                      hyf/no-commented-out-code

✖ 3 problems (2 errors, 1 warning)


2022-09-09 05:27:53 info: ---------------------------------------
2022-09-09 05:27:53 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:27:53 info: ---------------------------------------
2022-09-09 05:27:55 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ should call `new Promise()`
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument
    ✕ should resolve when the die settles successfully (1 ms)
    ✕ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should resolve when the die settles successfully

    TypeError: rollDie is not a function

      59 |       .mockImplementation((cb) => cb());
      60 |
    > 61 |     const promise = rollDie();
         |                     ^
      62 |     expect(promise).toBeInstanceOf(Promise);
      63 |     const assertionPromise = expect(promise).resolves.toBeDefined();
      64 |

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:61:21)

  ● rollDie › should resolve when the die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      50 |
      51 |   test('should resolve when the die settles successfully', () => {
    > 52 |     expect.assertions(3);
         |            ^
      53 |     expect(exported).toBeDefined();
      54 |
      55 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:52:12)

  ● rollDie › should reject with an Error when the die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      73 |
      74 |   test('should reject with an Error when the die rolls off the table', async () => {
    > 75 |     expect.assertions(3);
         |            ^
      76 |     expect(exported).toBeDefined();
      77 |
      78 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:75:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 5 passed, 7 total
Snapshots:   0 total
Time:        0.605 s, estimated 1 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-09-09 05:27:57 error: *** ESLint Report ***

ex3-rollDie.js
  18:11  error    'rollOnce' is assigned a value but never used  no-unused-vars
  43:10  error    'main' is defined but never used               no-unused-vars
  54:1   warning  Remove commented-out code                      hyf/no-commented-out-code

✖ 3 problems (2 errors, 1 warning)


2022-09-09 05:29:42 info: ---------------------------------------
2022-09-09 05:29:42 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:29:42 info: ---------------------------------------
2022-09-09 05:29:45 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ should call `new Promise()`
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✕ should resolve when the die settles successfully (1 ms)
    ✕ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should resolve when the die settles successfully

    TypeError: rollDie is not a function

      59 |       .mockImplementation((cb) => cb());
      60 |
    > 61 |     const promise = rollDie();
         |                     ^
      62 |     expect(promise).toBeInstanceOf(Promise);
      63 |     const assertionPromise = expect(promise).resolves.toBeDefined();
      64 |

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:61:21)

  ● rollDie › should resolve when the die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      50 |
      51 |   test('should resolve when the die settles successfully', () => {
    > 52 |     expect.assertions(3);
         |            ^
      53 |     expect(exported).toBeDefined();
      54 |
      55 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:52:12)

  ● rollDie › should reject with an Error when the die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      73 |
      74 |   test('should reject with an Error when the die rolls off the table', async () => {
    > 75 |     expect.assertions(3);
         |            ^
      76 |     expect(exported).toBeDefined();
      77 |
      78 |     const logSpy = jest.spyOn(console, 'log').mockImplementation();

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:75:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 5 passed, 7 total
Snapshots:   0 total
Time:        0.687 s, estimated 1 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-09-09 05:29:47 error: *** ESLint Report ***

ex3-rollDie.js
  43:10  error    'main' is defined but never used  no-unused-vars
  54:1   warning  Remove commented-out code         hyf/no-commented-out-code

✖ 2 problems (1 error, 1 warning)


2022-09-09 05:30:59 info: ---------------------------------------
2022-09-09 05:30:59 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:30:59 info: ---------------------------------------
2022-09-09 05:31:02 info: All unit tests passed.
2022-09-09 05:31:05 info: All steps were completed successfully
2022-09-09 05:38:44 info: ---------------------------------------
2022-09-09 05:38:44 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:38:44 info: ---------------------------------------
2022-09-09 05:38:47 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ should call `new Promise()`
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should resolve when the die settles successfully (2 ms)
    ✓ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.592 s, estimated 1 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-09-09 05:39:10 info: ---------------------------------------
2022-09-09 05:39:10 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-09-09 05:39:10 info: ---------------------------------------
2022-09-09 05:39:13 info: All unit tests passed.
2022-09-09 05:39:16 info: All steps were completed successfully
2022-09-09 07:16:27 info: --------------------------------------------
2022-09-09 07:16:27 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-09 07:16:27 info: --------------------------------------------
2022-09-09 07:16:31 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()` (1 ms)
    ✓ should use `Promise.all()`
    ✓ should resolve when all dice settle successfully (2 ms)
    ✕ should reject with an Error when a die rolls off the table (1 ms)

  ● ex4-pokerDiceAll › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 5 passed, 7 total
Snapshots:   0 total
Time:        0.887 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-09-09 07:29:20 info: --------------------------------------------
2022-09-09 07:29:20 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-09 07:29:20 info: --------------------------------------------
2022-09-09 07:29:23 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()` (1 ms)
    ✓ should use `Promise.all()`
    ✓ should resolve when all dice settle successfully (1 ms)
    ✕ should reject with an Error when a die rolls off the table

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.783 s, estimated 1 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-09-09 07:40:39 info: --------------------------------------------
2022-09-09 07:40:39 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-09 07:40:39 info: --------------------------------------------
2022-09-09 07:40:43 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()`
    ✓ should use `Promise.all()`
    ✕ should resolve when all dice settle successfully (1 ms)
    ✕ should reject with an Error when a die rolls off the table (1 ms)

  ● ex4-pokerDiceAll › should resolve when all dice settle successfully

    TypeError: rollDice is not a function

      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);
      52 |
    > 53 |     const promise = rollDice();
         |                     ^
      54 |     expect(promise).toBeInstanceOf(Promise);
      55 |     const result = await promise;
      56 |     expect(Array.isArray(result)).toBe(true);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:53:21)

  ● ex4-pokerDiceAll › should resolve when all dice settle successfully

    expect.assertions(4)

    Expected four assertions to be called but received one assertion call.

      46 |
      47 |   test('should resolve when all dice settle successfully', async () => {
    > 48 |     expect.assertions(4);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:48:12)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 5 passed, 7 total
Snapshots:   0 total
Time:        0.792 s, estimated 1 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
2022-09-09 07:40:45 error: *** ESLint Report ***

ex4-pokerDiceAll.js
   31:1   warning  Remove commented-out code         hyf/no-commented-out-code
   34:1   warning  Remove commented-out code         hyf/no-commented-out-code
   49:1   warning  Remove commented-out code         hyf/no-commented-out-code
   65:7   error    'underTest' is not defined        no-undef
   68:15  error    'moment' is not defined           no-undef
   77:22  error    'ROLL_TIME' is not defined        no-undef
  131:1   warning  Remove commented-out code         hyf/no-commented-out-code
  141:10  error    'main' is defined but never used  no-unused-vars
  147:1   warning  Remove commented-out code         hyf/no-commented-out-code

✖ 9 problems (4 errors, 5 warnings)


2022-09-09 07:42:29 info: --------------------------------------------
2022-09-09 07:42:29 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-09 07:42:29 info: --------------------------------------------
2022-09-09 07:42:33 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()`
    ✓ should use `Promise.all()` (1 ms)
    ✓ should resolve when all dice settle successfully (2 ms)
    ✕ should reject with an Error when a die rolls off the table (1 ms)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.672 s, estimated 1 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-09-09 07:45:48 info: --------------------------------------------
2022-09-09 07:45:48 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-09 07:45:48 info: --------------------------------------------
2022-09-09 07:45:51 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()` (1 ms)
    ✓ should use `Promise.all()` (1 ms)
    ✓ should resolve when all dice settle successfully (2 ms)
    ✕ should reject with an Error when a die rolls off the table (1 ms)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.736 s, estimated 1 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-09-09 07:45:54 error: *** Spell Checker Report ***

ex4-pokerDiceAll.js:48:73 - Unknown word (doesnt)

2022-09-09 07:48:52 info: ----------------------------------------------
2022-09-09 07:48:52 info: >>> Running Unit Test `ex5-pokerDiceChain` <<<
2022-09-09 07:48:52 info: ----------------------------------------------
2022-09-09 07:48:55 error: *** Unit Test Error Report ***

Command failed: npx jest ex5-pokerDiceChain.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex5-pokerDiceChain.test.js
  ex5-pokerDiceChain
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `rollDice` should not contain unneeded console.log calls (2 ms)
    ✓ should resolve when all dice settle successfully (53 ms)
    ✓ should reject with an Error when a die rolls off the table (28 ms)

  ● ex5-pokerDiceChain › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.<anonymous> (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        0.91 s
Ran all test suites matching /ex5-pokerDiceChain.test.js/i.
2022-09-16 06:13:21 info: ---------------------------------------
2022-09-16 06:13:21 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-09-16 06:13:21 info: ---------------------------------------
2022-09-16 06:13:25 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✕ should exist and be executable (4 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls (1 ms)
    ✕ should call `new Promise()` (1 ms)
    ✕ should take a single argument (2 ms)
    ✕ `resolve()` should be called with a one argument (1 ms)
    ✕ `reject()` should be called with a one argument (1 ms)
    ✕ should resolve when called with a string argument (1 ms)
    ✕ should reject with an Error object when called without an argument

  ● getAnonName › should exist and be executable

    expect(received).toBeDefined()

    Received: undefined

      46 |
      47 |   test('should exist and be executable', () => {
    > 48 |     expect(exported).toBeDefined();
         |                      ^
      49 |   });
      50 |
      51 |   testTodosRemoved(() => source);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:48:22)

  ● getAnonName › should call `new Promise()`

    expect(received).toBeDefined()

    Received: undefined

      54 |
      55 |   test('should call `new Promise()`', () => {
    > 56 |     expect(state.newPromise).toBeDefined();
         |                              ^
      57 |   });
      58 |
      59 |   test('should take a single argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:56:30)

  ● getAnonName › should take a single argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 0

      58 |
      59 |   test('should take a single argument', () => {
    > 60 |     expect(state.paramCount).toBe(1);
         |                              ^
      61 |   });
      62 |
      63 |   test('`resolve()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:60:30)

  ● getAnonName › `resolve()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      62 |
      63 |   test('`resolve()` should be called with a one argument', () => {
    > 64 |     expect(state.resolve).toBe(1);
         |                           ^
      65 |   });
      66 |
      67 |   test('`reject()` should be called with a one argument', () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:64:27)

  ● getAnonName › `reject()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      66 |
      67 |   test('`reject()` should be called with a one argument', () => {
    > 68 |     expect(state.reject).toBe(1);
         |                          ^
      69 |   });
      70 |
      71 |   test('should resolve when called with a string argument', async () => {

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:68:26)

  ● getAnonName › should resolve when called with a string argument

    expect(received).toBeDefined()

    Received: undefined

      71 |   test('should resolve when called with a string argument', async () => {
      72 |     expect.assertions(3);
    > 73 |     expect(exported).toBeDefined();
         |                      ^
      74 |     const timeoutSpy = jest
      75 |       .spyOn(global, 'setTimeout')
      76 |       .mockImplementation((cb) => cb());

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:73:22)

  ● getAnonName › should resolve when called with a string argument

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      70 |
      71 |   test('should resolve when called with a string argument', async () => {
    > 72 |     expect.assertions(3);
         |            ^
      73 |     expect(exported).toBeDefined();
      74 |     const timeoutSpy = jest
      75 |       .spyOn(global, 'setTimeout')

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:72:12)

  ● getAnonName › should reject with an Error object when called without an argument

    expect(received).toBeDefined()

    Received: undefined

      83 |   test('should reject with an Error object when called without an argument', async () => {
      84 |     expect.assertions(3);
    > 85 |     expect(exported).toBeDefined();
         |                      ^
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')
      88 |       .mockImplementation((cb) => cb());

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:85:22)

  ● getAnonName › should reject with an Error object when called without an argument

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      82 |
      83 |   test('should reject with an Error object when called without an argument', async () => {
    > 84 |     expect.assertions(3);
         |            ^
      85 |     expect(exported).toBeDefined();
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:84:12)

Test Suites: 1 failed, 1 total
Tests:       7 failed, 2 passed, 9 total
Snapshots:   0 total
Time:        1.119 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-09-16 06:13:26 error: *** ESLint Report ***

ex1-johnWho.js
  24:3  error  Parsing error: Missing catch or finally clause

✖ 1 problem (1 error, 0 warnings)


2022-09-16 06:23:07 info: ---------------------------------------
2022-09-16 06:23:07 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-09-16 06:23:07 info: ---------------------------------------
2022-09-16 06:23:11 info: All unit tests passed.
2022-09-16 06:23:15 info: All steps were completed successfully
2022-09-16 06:33:26 info: ---------------------------------------
2022-09-16 06:33:26 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-09-16 06:33:26 info: ---------------------------------------
2022-09-16 06:33:29 info: All unit tests passed.
2022-09-16 06:33:33 info: All steps were completed successfully
2022-09-16 06:38:02 info: --------------------------------------------
2022-09-16 06:38:02 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-16 06:38:02 info: --------------------------------------------
2022-09-16 06:38:04 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()`
    ✓ should use `Promise.all()`
    ✓ should resolve when all dice settle successfully (2 ms)
    ✕ should reject with an Error when a die rolls off the table (1 ms)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.698 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-09-16 06:38:28 info: --------------------------------------------
2022-09-16 06:38:28 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-16 06:38:28 info: --------------------------------------------
2022-09-16 06:38:31 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()` (1 ms)
    ✓ should use `Promise.all()`
    ✓ should resolve when all dice settle successfully (2 ms)
    ✕ should reject with an Error when a die rolls off the table (1 ms)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.49 s, estimated 1 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-09-16 06:38:45 info: --------------------------------------------
2022-09-16 06:38:45 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-16 06:38:45 info: --------------------------------------------
2022-09-16 06:38:48 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ `rollDice` should not contain unneeded console.log calls (1 ms)
    ✓ should use `dice.map()` (2 ms)
    ✓ should use `Promise.all()`
    ✓ should resolve when all dice settle successfully (2 ms)
    ✕ should reject with an Error when a die rolls off the table (1 ms)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.593 s, estimated 1 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (/Users/hackyourfuture/Desktop/hyf-course/repos/Homework-class39/3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-09-16 06:46:35 info: --------------------------------------------
2022-09-16 06:46:35 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-09-16 06:46:35 info: --------------------------------------------
2022-09-16 06:46:39 info: All unit tests passed.
2022-09-16 06:46:43 info: All steps were completed successfully
2022-09-16 07:17:00 info: ----------------------------------------------
2022-09-16 07:17:00 info: >>> Running Unit Test `ex5-pokerDiceChain` <<<
2022-09-16 07:17:00 info: ----------------------------------------------
2022-09-16 07:17:04 info: All unit tests passed.
2022-09-16 07:17:09 info: All steps were completed successfully
